src/App.js

```diff
@@ -11,23 +18,82 @@ class App extends React.Component {
       cardDeck: makeShuffledDeck(),
       // currCards holds the cards from the current round
       currCards: [],
+      player1score: 0,
+      player2score: 0,
+      isCardDeckEmpty: false,
+      pastRounds: [],
     };
   }
 
+  gameRestarted = () => {
+    this.setState((state) => ({
+      isCardDeckEmpty: false,
+      cardDeck: makeShuffledDeck(),
+      currCards: [],
+      pastRounds: [
+        ...state.pastRounds,
+        [state.player1score, state.player2score],
+      ],
+      player1score: 0,
+      player2score: 0,
+    }));
+
+    console.log(this.state.pastRounds);
+  };
+
   dealCards = () => {
     this.setState((state) => ({
       // Remove last 2 cards from cardDeck
       cardDeck: state.cardDeck.slice(0, -2),
       // Deal last 2 cards to currCards
       currCards: state.cardDeck.slice(-2),
     }));
+
+    // Add score to the player that wins the round. No points for draw.
+    if (this.state.currCards[0].rank > this.state.currCards[1].rank) {
+      this.setState((state) => ({ player1score: state.player1score + 1 }));
```

Comment:
the issue you have with the score not updating on click is that your state for player score is dependent on the state for `currCards`, but `currCards` does not update until `dealCards` has finished executing. 

to overcome this problem, you will need to update all the states at once - i.e. both players scores and the `isCardDeckEmpty` property. refer to the model solution: https://github.com/rocketacademy/high-card-bootcamp/blob/f751025e97f819e4dbe12db31a73c922dbf8cc91/src/App.js#L42

alternatively, if you're learning the hooks version of react, you can make use of `useEffect`s to listen to changes in the state

Reply:
I see. It works now.